lucetc/src/decls.rs

```diff
@@ -267,25 +307,42 @@ impl<'a> ModuleDecls<'a> {
         for ix in 0..info.globals.len() {
             let ix = GlobalIndex::new(ix);
             let g_decl = info.globals.get(ix).unwrap();
-            let g_import = info.imported_globals.get(ix);
-            let g_variant = if let Some((module, field)) = g_import {
-                GlobalVariant::Import { module, field }
-            } else {
-                let init_val = match g_decl.entity.initializer {
-                    // Need to fix global spec in ModuleData and the runtime to support more:
-                    GlobalInit::I32Const(i) => i as i64,
-                    GlobalInit::I64Const(i) => i,
-                    _ => Err(format_err!(
-                        "non-integer global initializer: {:?}",
```

Comment:
Supporting float globals gets us passing the `global` spectest, though this PR doesn't add import global support into `lucet-runtime` (yet?), which is the entire reason I started changing globals here! :grimacing: